plugins/coverbrowser.koplugin/mosaicmenu.lua

```diff
@@ -648,6 +652,44 @@ function MosaicMenuItem:update()
                 self.has_description = true
             end
 
+            if not self.menu.cover_info_cache then
+                self.menu.cover_info_cache = {}
+            end
+
+            -- Current page / pages are available or more accurate in .sdr/metadata.lua
+            -- We use a cache (cleaned at end of this browsing session) to store
+            -- page, percent read and book status from sidecar files, to avoid
+            -- re-parsing them when re-rendering a visited page
```

Comment:
May be add this comment:
This cache is shared with ListMenu, so we need to fill it with the same info here than there, even if we don't need them all here.